Src/FluentAssertions/Execution/AssertionScope.cs

```diff
@@ -29,6 +29,18 @@ public sealed class AssertionScope : IAssertionScope
         private string fallbackIdentifier = "object";
         private bool? succeeded;
 
+        private sealed class DeferredReportable
+        {
+            private readonly Lazy<string> lazyValue;
```

Comment:
Is there a known case where `valueFunc` is invoked more than once?

Reply:
We're only calling `ContextDateItems.GetReportable` when the scope is disposed, and only if it's the outer scope, so I think we're fine.